operator/controllers/btpoperator_controller.go

```diff
@@ -978,38 +1122,67 @@ func (r *BtpOperatorReconciler) deleteAllOfinstalledResources(ctx context.Contex
 	return nil
 }
 
-func (r *BtpOperatorReconciler) HandleReadyState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
+func (r *BtpOperatorReconciler) prepareInstallInfo(ctx context.Context, cr *v1alpha1.BtpOperator) (types.InstallInfo, Reason, error) {
 	logger := log.FromContext(ctx)
-	logger.Info("Handling Ready state")
 
 	secret, err := r.getRequiredSecret(ctx)
 	if err != nil {
 		logger.Error(err, "while getting the required Secret")
-		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, MissingSecret, "Secret resource not found")
+		return types.InstallInfo{}, MissingSecret, fmt.Errorf("Secret resource not found")
 	}
 
 	if err = r.verifySecret(secret); err != nil {
 		logger.Error(err, "while verifying the required Secret")
-		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, InvalidSecret, "Secret validation failed")
+		return types.InstallInfo{}, InvalidSecret, fmt.Errorf("Secret validation failed")
 	}
 
 	r.addTempLabelsToCr(cr)
 
 	installInfo, err := r.getInstallInfo(ctx, cr, secret)
 	if err != nil {
 		logger.Error(err, "while preparing InstallInfo")
-		return err
+		return types.InstallInfo{}, PreparingInstallInfoFailed, err
 	}
 	if installInfo.ChartPath == "" {
-		return fmt.Errorf("no chart path available for processing")
+		return types.InstallInfo{}, ChartPathEmpty, fmt.Errorf("no chart path available for processing")
+	}
+
+	return installInfo, "", nil
+}
+
+func (r *BtpOperatorReconciler) doConsistencyCheck(ctx context.Context, cr *v1alpha1.BtpOperator) (bool, Reason, error) {
```

Comment:
why does the method return `bool` value? Also, you can set Reason directly on the custom resource passed as one of the args